src/util/types.ts

```diff
@@ -42,10 +42,31 @@ export interface RelayerState {
     readonly selectedToken: Token | null;
 }
 
+export enum StepKind {
+    BuySellLimit,
+}
+
+export interface StepBuySellLimitOrder {
+    kind: StepKind.BuySellLimit;
+    amount: BigNumber;
+    price: BigNumber;
+    side: OrderSide;
+}
+
+export type Step = StepBuySellLimitOrder;
+
+export interface StepsModalState {
+    readonly isVisible: boolean;
+    readonly doneSteps: Step[];
+    readonly currentStep: Step | null;
+    readonly pendingSteps: Step[];
+}
+
 export interface StoreState {
     readonly router: RouterState;
     readonly blockchain: BlockchainState;
     readonly relayer: RelayerState;
+    readonly stepsModal: StepsModalState;
```

Comment:
I'm working on the notifications feature, and I added a `UIState` that holds the part of the global state that is related to the UI. Do you mind doing it that way so we don't have conflicts here? I guess stepsModal should be a property in `UIState`.